src/backend/WorkflowManagement/WorkflowManager.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading.Tasks;
+
+namespace WorkflowManagement
+{
+    public enum WorkflowStage
+    {
+        Analyze,
+        Report,
+        Telemetry,
+        Finished
+    }
+
+    /// <summary>
+    /// Workflow stages are in the following order: Analyze -> Report -> Telemetry -> Finished
+    /// </summary>
+    public class WorkflowManager
+    {
+        readonly IWorkflowAction[] actions;
+        static WorkflowManager manager;
+
+        public WorkflowManager()
+        {
+            IWorkflowAction[] workflowActions = new IWorkflowAction[3];
+
+            workflowActions[(int)WorkflowStage.Analyze] = new AnalyzeAction();
+            workflowActions[(int)WorkflowStage.Report] = new ReportAction();
+            workflowActions[(int)WorkflowStage.Telemetry] = new TelemetryAction();
+
+            actions = workflowActions;
+        }
+
+        public WorkflowManager(IWorkflowAction[] workflowActions)
+        {
+            actions = workflowActions;
+        }
+
+        /// <summary>
+        /// Gets the first message used to start a workflow for a submission.
+        /// </summary>
+        public WorkflowQueueMessage GetFirstStage(string submissionId)
+        {
+            return new WorkflowQueueMessage() { SubmissionId = submissionId, Stage = WorkflowStage.Analyze };
+        }
+
+        /// <summary>
+        /// Executes the action for the current stage in the workflow. After the action completes,
+        /// returns a message for the next stage in the queue. 
+        /// </summary>
+        public async Task<WorkflowQueueMessage> ExecuteActionsToNextStage(WorkflowQueueMessage currentMsg)
+        {
+            //Execute the action for the current stage
+            await actions[(int)currentMsg.Stage].Execute(currentMsg.SubmissionId);
+
+            //Determine which stage in the workflow the submission moves to next
+            WorkflowStage nextStage;
+            switch (currentMsg.Stage)
```

Comment:
Similar to my last comment, I wouldn't include this switch statement. Instead, I think `IWorkflowAction.Execute` should return a `Task<WorkflowStage>` indicating which stage comes next. That way it's the *workflow stage's* responsibility to define the flow, the workflow manager just follows it. It will also allow for more complex scenarios in the future where we don't execute each stage linearly. For example, if the analysis stage failed, it could report that next is the 'ErrorHandling' stage or something instead of moving onto the report stage like usual.